run-go-mod-tidy.sh

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+if go mod tidy -v 2>&1 | grep -q 'updates to go.mod needed'; then
+    exit 1
```

Comment:
What is the objective of this hook? Is it to ensure that the `go.mod` file is always tidy before committing? If that's the case, would something like this work instead of `grep` ?

```
git diff --exit-code go.*
```

That should report if any changes were made by `go mod tidy`, and cause the hook to fail.

Reply:
We can't use `git diff --exit-code` because `go.sum` and `go.mod` can have differences even after running `go mod tidy`.

For example if there is new dependencies they can be re-ordered by the command, the `git diff` would make the hook crash after legitimate changes.

What do you think about that?

Reply:
That makes sense, I guess `git diff --exit-code` does not work for this.